pkg/templates/upgradeconfig/template.go

```diff
@@ -0,0 +1,161 @@
+package upgradeconfig
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"golang.stackrox.io/kube-linter/internal/stringutils"
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/extract"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/upgradeconfig/internal/params"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	templateKey = "upgrade-configuration"
+)
+
+func compareIntOrString(max, min string, actual *intstr.IntOrString) bool {
+	if len(max) == 0 && len(min) == 0 {
+		return true
+	}
+	if actual == nil {
+		return false
+	}
+	actualVal, err := intstr.GetValueFromIntOrPercent(actual, 100, false)
+	if err != nil {
+		return false
+	}
+	actualIsPercent := strings.Contains(actual.String(), "%")
+	if len(max) > 0 {
+		maxIntOrString := intstr.Parse(max)
+		maxIsPercent := strings.Contains(maxIntOrString.String(), "%")
+		if actualIsPercent != maxIsPercent {
```

Comment:
Hmm, why?

Reply:
So, as I was thinking about how to handle percent vs fixed, I came up with a few options:
1) Grab the replicas  and do a calculation to determine if a fix MaxPodsUnavailable would meet a percent requirement in the check
2) Force the two entities to be the same type

I went with 2 because while 1 would work for an initial check, it doesn't allow for scaling once deployed.  A fixed number of MaxUnavailable may meet a percent requirement with the initial deployment, but then would not meet it if the deployment is scaled after deployment.  So, I decided if the check is using a percent value then a percent should be enforced on the entity being checked to ensure it stays correct even if the replicas are scaled after deployment